functorch/csrc/DynamicLayer.h

```diff
@@ -45,6 +46,9 @@ struct TORCH_API DynamicLayer {
 
   // only valid for jvp transform
   optional<bool> prevFwdGradMode() const;
+
+  // only valid for functionalization
+  optional<bool> functionalizeAddBackViews() const;
```

Comment:
Lol I am going to turn these optional fields into a Union after we merge this, I didn't expect us to add so many